src/sinks/gcp_pubsub/run.rs

```diff
@@ -4,44 +4,28 @@ use async_recursion::async_recursion;
 use cloud_pubsub::{error::Error, Client, Topic};
 use serde_json::json;
 
-use crate::{model::Event, pipelining::StageReceiver, sinks::ErrorPolicy, utils::Utils};
+use crate::{
+    model::Event,
+    pipelining::StageReceiver,
+    sinks::ErrorPolicy,
+    utils::{retry, Utils},
+};
 
 #[async_recursion]
-async fn send_pubsub_msg(
-    client: &Topic,
-    event: &Event,
-    policy: &ErrorPolicy,
-    retry_quota: usize,
-    backoff_delay: Duration,
-) -> Result<(), Error> {
+async fn send_pubsub_msg(client: &Topic, event: &Event) -> Result<(), Error> {
     let body = json!(event).to_string();
 
-    let result = client.publish(body).await;
+    client.publish(body).await?;
 
-    match (result, policy, retry_quota) {
-        (Ok(_), _, _) => {
-            log::info!("successful pubsub publish");
-            Ok(())
-        }
-        (Err(x), ErrorPolicy::Exit, 0) => Err(x),
-        (Err(x), ErrorPolicy::Continue, 0) => {
-            log::warn!("failed to publish to pubsub: {:?}", x);
-            Ok(())
-        }
-        (Err(x), _, quota) => {
-            log::warn!("failed attempt to execute pubsub publish: {:?}", x);
-            std::thread::sleep(backoff_delay);
-            send_pubsub_msg(client, event, policy, quota - 1, backoff_delay).await
-        }
-    }
+    Ok(())
 }
 
 pub fn writer_loop(
     input: StageReceiver,
     credentials: String,
     topic_name: String,
     error_policy: &ErrorPolicy,
-    max_retries: usize,
+    max_retries: u32,
     backoff_delay: Duration,
```

Comment:
we could do it in another PR, but it would be nice to pass the RetryPolicy directly. We provide more options and also reduce the number of parameters in the call.

Reply:
cool I considered that, I'll just switch to that